Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptSearch.java

```diff
@@ -55,29 +60,53 @@
+    public @ResponseBody ResponseEntity<String> searchConcept(
+            @Valid ConceptSearchParameters conceptSearchParameters)
+                    throws IllegalArgumentException, IllegalAccessException {
         Map<String, String> searchFields = new HashMap<String, String>();
         String operator = SearchParamters.OP_OR;
-        for (String key : queryParams.keySet()) {
-            if (key.trim().equals(SearchParamters.OPERATOR) && !queryParams.get(key)[0].trim().isEmpty()) {
-                operator = queryParams.get(key)[0].trim();
-            } else if (key.trim().equalsIgnoreCase(SearchFieldNames.TYPE_URI)) {
-                searchFields.put("type_id", uriHelper.getTypeId(queryParams.get(key)[0]));
-            } else {
-                searchFields.put(key.trim(), queryParams.get(key)[0]);
+        int page = 1;
+
+        for (Field field : conceptSearchParameters.getClass().getDeclaredFields()) {
+            field.setAccessible(true);
+            if (field.getName().equalsIgnoreCase("type_uri")) {
+                searchFields.put("type_id",
+                        uriHelper.getTypeId(String.valueOf(field.get(conceptSearchParameters))));
+            } else if (SearchParamters.OPERATOR.equalsIgnoreCase(field.getName())) {
+                operator = String.valueOf(field.get(conceptSearchParameters));
+            } else if (SearchParamters.PAGE.equalsIgnoreCase(field.getName())) {
+                page = field.get(conceptSearchParameters) != null
+                        ? (Integer) field.get(conceptSearchParameters) : page;
+            } else if (SearchParamters.NUMBER_OF_RECORDS_PER_PAGE
+                    .equalsIgnoreCase(field.getName())) {
+                numberOfRecordsPerPage = field.get(conceptSearchParameters) != null
+                        ? (Integer) field.get(conceptSearchParameters) : numberOfRecordsPerPage;
+            } else if (field.get(conceptSearchParameters) != null) {
+                searchFields.put(field.getName().trim(), String.valueOf(field.get(conceptSearchParameters)));
             }
         }
+
         ConceptEntry[] searchResults = null;
 
+        int totalNumberOfRecords = 0;
         try {
-            searchResults = manager.searchForConcepts(searchFields, operator);
+            totalNumberOfRecords = manager.getTotalNumberOfRecordsForSearch(searchFields, operator);
+            searchResults = manager.searchForConceptByPageNumberAndFieldMap(searchFields, operator,
+                    page, numberOfRecordsPerPage);
         } catch (LuceneException | IllegalAccessException | IndexerRunningException ex) {
             return new ResponseEntity<String>(ex.getMessage(), HttpStatus.BAD_REQUEST);
         }
 
+        if (searchResults.length == 0) {
+            // Returning a 404 error.
+            return new ResponseEntity<String>("No records found for the search condition.",
+                    HttpStatus.NOT_FOUND);
```

Comment:
shouldn't be not found. Not found would mean the resource wasn't found, but it was. It just didn't yield any results, which is just as ok as results.